src/contracts/proxies/actions/CommonActions.sol

```diff
@@ -56,8 +56,15 @@ abstract contract CommonActions is ICommonActions {
     address _systemCoin = address(ICoinJoin(_coinJoin).systemCoin());
     // Transfers coins from the user to the proxy
     SafeERC20.safeTransferFrom(IERC20(_systemCoin), msg.sender, address(this), _wad);
-    // Approves adapter to take the COIN amount
-    SafeERC20.safeApprove(IERC20(_systemCoin), _coinJoin, _wad);
+    //check if coinJoin has an approval amount
+    uint256 allowance = IERC20(_systemCoin).allowance(address(this), _coinJoin);
+    if (allowance == 0) {
+      // Approves adapter to take the COIN amount
+      SafeERC20.safeApprove(IERC20(_systemCoin), _coinJoin, _wad);
+    } else {
+      // if approval amount is a non-zero number, increase allowance.
+      SafeERC20.safeIncreaseAllowance(IERC20(_systemCoin), _coinJoin, _wad);
```

Comment:
I can't think of any security implications here. If the tx fails, then the allowance isn't increased, therefore there is no risk of having allowance higher than necessary.

Reply:
that's up to you.  we could make it to check and modify the allowance to only be enough to allow the TX in question if that's the more desired solution.  e.g. 
```if(allowance < desiredAllowance) {
_wad = desiredAllowance - allowance;
SafeERC20.safeIncreasAllowance(IERC20(_systemCoin), _coinJoin, _wad);
} ```